src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java

```diff
@@ -118,6 +144,191 @@ public static String getBlobEP(String blobURL) {
+            credentialId.appendChild(document.createTextNode(storageCredId));
+            node.appendChild(credentialId);
+
+            // write the content into job config xml file
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            DOMSource source = new DOMSource(document);
+            StreamResult result = new StreamResult(new File(jobConfigFile));
+            transformer.transform(source, result);
+        }
+    }
+
+    public static void writeCredentialIdInJob(String jobName, String storageCredId) throws SAXException, IOException, ParserConfigurationException, TransformerConfigurationException, TransformerException {
+        String jobDirectory = getWorkDirectory() + "\\jobs\\" + jobName;
+        String jobConfigFile = jobDirectory + "\\" + "config.xml";
+
+        DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+        Document document = docBuilder.parse(jobConfigFile);
+        NodeList nodeList = null;
+        
+        Utils.backupFile(jobConfigFile); //backup jobconfigFile    
+        
+        if (document.getElementsByTagName("storageCredentialId").getLength() <= 0) {
+            //update download job config
+            nodeList = document.getElementsByTagName("com.microsoftopentechnologies.windowsazurestorage.AzureStorageBuilder");
+            writeXMLElements(jobConfigFile, document, nodeList, storageCredId);
+            //update upload job config
+            nodeList = document.getElementsByTagName("com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher");
+            writeXMLElements(jobConfigFile, document, nodeList, storageCredId);
+        
+        } else { //job already upgraded and has credentials-id
+            return;
+        }
+
+    }
+
+    public static void backupFile(String sourceFile) throws IOException {
+        String backupFile = sourceFile + ".backup";
+
+        FileUtils.copyFile(new File(sourceFile), new File(backupFile));
+    }
+    
+    /**
+     *
+     * Take the legacy local storage credential configuration and create an
+     * equivalent global credential in Jenkins Credential Store
+     *
+     */
+    public static void removeFile(String sourceFile) throws IOException {
+        File file = new File(sourceFile);
+
+        if (file.delete()) {
+            System.out.println(file.getName() + " is deleted!");
+        } else {
+            System.out.println("Delete operation is failed.");
+        }
+    }
+
+    public static void upgradeStorageConfig() throws Exception {
+        String sourceFile = getWorkDirectory() + "\\" + Constants.LEGACY_STORAGE_CONFIG_FILE;
+        try {
+            //check if it has been upgraded already
+            File upgradeFile = new File(sourceFile+".backup");
```

Comment:
shouldn't the existence of a backup file mean that the previous migration attempt wasn't successful and we should try again?